### DISCUSSION TITLE
Statamic updater can't find php locally

### DISCUSSION THREAD

**Discussion by than** [statamic/cms#5974] [Troubleshooting]
Updated all the things, composer included and still getting 
`sh: line 0: exec: : not found`
whenever I attempt to trigger updates through the CP 

Similar issue was mentioned here, but I'm not seeing any resolution. https://statamic.com/forum/5412-issue-with-updates-in-version-3-1-11

**Comment by edalzell** [comment]
How do you know it's php that can't be found? I don't see that mentioned in the error.

Is there more to that error somewhere?

In the meantime, update via the command line, that's what I do on all our sites.

**Reply by than** [reply]
Using valet, fwiw. I should've added it apparently can't `exec`. 
`sh: line 0: exec: : not found` is the entirety of the error, and doesn't provide anything else. 
Local logs aren't any more verbose:
```
[2022-05-02 18:07:31] local.ERROR: Composer Process: sh: line 0: exec: : not found
```

**Comment by jackmcdade** [comment]
Try deleting the `vendor` and `composer-lock.json` files and running `composer install` again.

**Reply by than** [reply]
No dice. Same error output.

**Comment by jasonvarga** [comment]
Try running `composer update`

I think this has to do with an issue I found in Composer in March which has since been fixed. https://github.com/composer/composer/issues/10662

**Reply by jesseleite** [reply]
I think you mean `composer self-update`?

**Reply by jasonvarga** [reply]
Nope I mean `composer update` - Statamic requires `composer/composer`. Composer updating will update composer. 🤯

**Reply by jesseleite** [reply]
Oh THAT composer, yep makes sense. It's late here :D
